README.md

```diff
@@ -1,33 +1,103 @@
+# Installing Build Dependencies
 
-# Build the cross-compiler
+The [main build script](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/build.sh)
+installs its dependencies automatically when the `RUN_BOOTSTRAP=1` environment variable is defined,
+which is the default. To see what will be installed refer to
+[`.github/scripts/install-dependencies.sh`](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/.github/scripts/install-dependencies.sh).
 
-Running build.sh will build the cross compiler. It installs it into ~/cross. The build script does multiple things.
- - Downloads dependencies
- - Clones several dependencies 
-    - modified binutils from https://github.com/Windows-on-ARM-Experiments/binutils-woarm64.git
-    - modified gcc from https://github.com/Windows-on-ARM-Experiments/gcc-woarm64
-    - modified mingw from https://github.com/Windows-on-ARM-Experiments/mingw-woarm64
- - Builds and installs: binutils, mingw, gcc and libgcc.
+# Building the Cross-Compiler
 
-# Using the cross compiler
+To build the cross-compiler and install it into a `~/cross-aarch64-w64-mingw32-msvcrt` folder,
+follow these steps:
 
-To build a simple C file, something like this works:
+1. Clone the repository:
+   ```bash
+   git clone https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build.git
+   ```
+
+2. Navigate to the root repository folder:
+   ```bash
+   cd mingw-woarm64-build
+   ```
+
+3. Run the build script:
+   ```bash
+   ./build.sh
+   ```
+
+To install the toolchain into a different folder you can use the `TOOLCHAIN_PATH` environment
+variable:
+```bash
+TOOLCHAIN_PATH=/opt/aarch64-w64-mingw32 ./build.sh
 ```
-~/cross/bin/aarch64-w64-mingw32-gcc hello.c
+
+To skip the dependencies bootstrapping process and fetching source code repositories for consecutive
+builds you can use other environment variables:
+```bash
+RUN_BOOTSTRAP=0 UPDATE_SOURCES=0 ./build.sh
 ```
-I was able to cross compile zlib and install as follows:
+
+The build script does multiple things:
+
+- Installs the dependency Ubuntu packages.
+- Clones several dependency source code repositories:
+  - modified binutils from [Windows-on-ARM-Experiments/binutils-woarm64](https://github.com/Windows-on-ARM-Experiments/binutils-woarm64),
+  - modified GCC from [Windows-on-ARM-Experiments/gcc-woarm64](https://github.com/Windows-on-ARM-Experiments/gcc-woarm64),
+  - modified MinGW from [Windows-on-ARM-Experiments/mingw-woarm64](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64).
+- Bootstraps the binutils and GCC sources with `gmp`, `mpfr`, `isl`, etc. dependencies.
+- Builds and installs binutils, stage 1 GCC, MinGW runtimes, GCC with libgcc, and then
+  the entire MinGW.
+
+# Using the Cross-Compiler
+
+After building the toolchain, to build a simple C source code file run:
+```bash
+PATH="~/cross-aarch64-w64-mingw32-msvcrt/bin:$PATH" aarch64-w64-mingw32-gcc hello.c -o hello.exe
```

Comment:
I think it is useful to enable the toolchain by extending PATH in a separate command.

Reply:
Changed.